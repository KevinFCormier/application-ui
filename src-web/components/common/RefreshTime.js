/*******************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2018, 2019. All Rights Reserved.
 *
 * Note to U.S. Government Users Restricted Rights:
 * Use, duplication or disclosure restricted by GSA ADP Schedule
 * Contract with IBM Corp.
 *******************************************************************************/

import React from 'react'
import PropTypes from 'prop-types'
import { Loading } from 'carbon-components-react'
import moment from 'moment'


class RefreshTime extends React.PureComponent {
  static propTypes = {
    refetch: PropTypes.func.isRequired,
    reloading: PropTypes.bool,
    timestamp: PropTypes.string.isRequired,
  }

  handleClick = () => {
    this.props.refetch()
  }

  render() {
    const { reloading, timestamp } = this.props
    const time = moment(new Date(timestamp)).format('h:mm:ss A')
    return (
      <div className='refresh-time-container'>
        {reloading ?<Loading withOverlay={false} small /> : null }
        <div>{time}</div>
      </div>
    )
  }
}

export default RefreshTime
