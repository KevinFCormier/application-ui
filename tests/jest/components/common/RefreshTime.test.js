/*******************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2019. All Rights Reserved.
 *
 * Note to U.S. Government Users Restricted Rights:
 * Use, duplication or disclosure restricted by GSA ADP Schedule
 * Contract with IBM Corp.
 *******************************************************************************/
"use strict";

import React from "react";
import RefreshTime from "../../../../src-web/components/common/RefreshTime";
import renderer from "react-test-renderer";

describe("RefreshTime component", () => {
  it("renders as expected", () => {
    const component = renderer.create(<RefreshTime />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  it("renders as expected", () => {
    const props = {
      reloading: true,
      timestamp: "03:34:07",
      refetch() {}
    };
    const component = renderer.create(<RefreshTime {...props} />);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
