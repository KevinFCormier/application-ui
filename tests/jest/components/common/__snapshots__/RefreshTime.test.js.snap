// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RefreshTime component renders as expected 1`] = `
<div
  className="refresh-time-container"
>
  <div>
    Invalid date
  </div>
</div>
`;

exports[`RefreshTime component renders as expected 2`] = `
<div
  className="refresh-time-container"
>
  <div
    aria-live="assertive"
    className="bx--loading bx--loading--small"
  >
    <svg
      className="bx--loading__svg"
      viewBox="-75 -75 150 150"
    >
      <circle
        cx="0"
        cy="0"
        r="37.5"
      />
    </svg>
  </div>
  <div>
    Invalid date
  </div>
</div>
`;
